R/tbl_listing.R

```diff
@@ -0,0 +1,171 @@
+#'   columns to be sorted. It defaults to `keys`.
+#' @param blank_rows_by ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
+#'   columns where changing values is highlighted by a blank row. It depends substantially on the columns'
+#'   sorting. See [crane::add_blank_row()] for more information. Defaults to `NULL`.
+#' @param hide_duplicate_keys (`logical`)\cr
+#'   whether to add blank values where key columns have duplicate values. Defaults to `TRUE`.
+#'
+#' @note
+#' Common pre-processing steps for the data frame that may be common:
+#'  * Unique values - this should be enforced in pre-processing by users.
+#'  * `NA` values - they are not printed by default in `{gtsummary}`. You can make them explicit if
+#'    they need to be displayed in the listing. See example 3.
+#'  * Split by rows - you can split the data frame by rows and then apply `tbl_listing()` to each subset.
+#'    See example 5.
+#'  * Split by columns - you can split the data frame by columns and then apply `tbl_listing()` to each subset.
+#'    See example 6.
+#'  * Split in post-processing is not suggested if `hide_duplicate_keys = TRUE`.
+#'
+#' @examplesIf rlang::is_installed("labelled")
+#' # Load the trial dataset
+#' trial_data <- trial |>
+#'   dplyr::select(trt, age, marker, stage) |>
+#'   dplyr::filter(stage %in% c("T2", "T3")) |>
+#'   dplyr::slice_head(n = 2, by = c(trt, stage)) # downsampling
+#'
+#' # Example 1 --------------------------------
+#' tbl_listing(trial_data, keys = c(trt, stage))
+#'
+#' # Example 2 --------------------------------
+#' # Sort by trt stage and marker
+#' tbl_listing(trial_data, keys = c(trt, stage), order_by = c(trt, stage, marker))
+#'
+#' # Example 3 --------------------------------
+#' # make NAs explicit
+#' trial_data_na <- trial_data |>
+#'   mutate(across(everything(), ~ tidyr::replace_na(labelled::to_character(.), "-")))
+#' tbl_listing(trial_data_na, keys = c(trt, stage))
+#'
+#' # Example 4 --------------------------------
+#' # Add blank rows for first key column
+#' lst <- tbl_listing(trial_data_na, keys = c(trt, stage), blank_rows_by = trt)
+#' lst
+#'
+#' # Can add them also manually in post-processing
+#' lst |> add_blank_row(row_numbers = seq(2))
+#'
+#' # Example 5 --------------------------------
+#' # Split by rows
+#' trial_data_split <- trial_data |>
+#'   split(trial_data$trt)
+#' list_lst <- lapply(trial_data_split, tbl_listing, keys = c(trt, stage))
+#' # names(list_lst) # keeps names
+#' list_lst[[2]]
+#'
+#' # Example 6 --------------------------------
+#' # Split by columns
+#' column_groups <- list(
+#'   age = c("trt", "age"),
+#'   marker = c("trt", "marker")
+#' )
+#' trial_data_split <- lapply(column_groups, function(cols) trial_data[, cols, drop = FALSE])
+#' list_lst <- lapply(trial_data_split, tbl_listing, keys = trt)
+#' # names(list_lst) # keeps names
+#' list_lst[[2]]
+#'
+#' @export
+tbl_listing <- function(data,
+                        keys = NULL,
+                        order_by = keys,
```

Comment:
This should fix your tidyselect external vector warning :)

```suggestion
                        order_by = all_of(keys),
```